app/assets/javascripts/components/general_button.es6.jsx

```diff
@@ -43,12 +57,31 @@ class GeneralButton extends React.Component {
     }
   }
 
+  handleMouseDown(event) {
+    this.setState({ hover: true });
+  }
+
+  handleMouseUp(event) {
+    this.setState({ hover: false });
+  }
+
+  handleMouseLeave(event) {
+    if (this.state.hover) {
+      this.setState({ hover: false });
+    }
+  }
+
   render() {
+    var style = Object.assign(
+      this.styles.container,
+      this.state.hover && this.styles.hover
+    );
```

Comment:
`this.state.hover && this.styles.hover` is taking advantage of the short-circuit nature of an `&&` conditional. If the component's current `hover` state is not true, then the conditional will short circuit and `this.styles.hover` (a style defined above) will not be merged with `var style`. If the component's current `hover` state is true, then the conditional will not short circuit and the last true object of the conditional will be returned, merging `this.styles.hover` into `var style`.